.github/actions/run-backend-tests/action.yml

```diff
@@ -63,12 +69,28 @@ runs:
         - uses: syphar/restore-pip-download-cache@v1
           if: steps.cache-backend-tests.outputs.cache-hit != 'true'
 
-        - name: Install python dependencies
+        - name: Install Python dependencies
           if: steps.cache-backend-tests.outputs.cache-hit != 'true'
           shell: bash
           run: |
-              python -m pip install -r requirements-dev.txt
-              python -m pip install -r requirements.txt
+              pip install -r requirements.txt -r requirements-dev.txt
+
+        - name: Install the working version of hogql-parser
+          if: steps.hogql-parser-diff.outputs.changed == 'true'
+          shell: bash
+          # This is not cached currently, as it's important to build the current HEAD version of hogql-parser if it has
+          # changed (requirements.txt has the already-published version)
+          run: |
+              sudo apt-get install libboost-all-dev unzip cmake curl uuid pkg-config
+              curl https://www.antlr.org/download/antlr4-cpp-runtime-4.13.0-source.zip --output antlr4-source.zip
```

Comment:
Might be worth chekcing the `md5sum` here?

Reply:
Good idea, added that!